clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -316,24 +312,26 @@ basic_counts_t::generate_interval_snapshot(uint64_t interval_id)
 {
     count_snapshot_t *snapshot = new count_snapshot_t;
     for (const auto &shard : shard_map_) {
-        counters_t shard_total;
         for (const auto &ctr : shard.second->counters) {
-            shard_total += ctr;
+            snapshot->counters += ctr;
         }
-        snapshot->delta_counters += shard_total;
-        snapshot->delta_counters -= shard.second->counters_at_last_interval;
-        shard.second->counters_at_last_interval = shard_total;
     }
     return snapshot;
 }
 
 analysis_tool_t::interval_state_snapshot_t *
-basic_counts_t::combine_interval_snapshot(analysis_tool_t::interval_state_snapshot_t *one,
-                                          analysis_tool_t::interval_state_snapshot_t *two)
+basic_counts_t::combine_interval_snapshots(
+    const std::vector<const analysis_tool_t::interval_state_snapshot_t *>
+        last_shard_snapshots,
+    const std::vector<bool> observed_in_cur_interval)
 {
     count_snapshot_t *result = new count_snapshot_t;
-    result->delta_counters = dynamic_cast<count_snapshot_t *>(one)->delta_counters;
-    result->delta_counters += dynamic_cast<count_snapshot_t *>(two)->delta_counters;
+    for (const auto snapshot : last_shard_snapshots) {
+        if (snapshot == nullptr)
+            continue;
+        result->counters +=
+            dynamic_cast<const count_snapshot_t *const>(snapshot)->counters;
```

Comment:
Wait, it's printing the detla below, so doesn't it want to only add if observed_in_cur_interval?  Worth adding a comment too explicitly stating which way it's going.